Failure to build from source

For some reason, the build process fails. Las year, I was able to build v7.0 without issues. Now, trying the same for v7.1, the build process fails with an error.
Now, of course I tried v7.0 again, to make sure it's not v7.1. Turns out, v7.0 now fails to build too.

Last year, the current versions of gcc and cmake on my distro were 13.2 and 3.30.1, respectively. Now, the versions are 14.2.1 and 4.1.1, respectively. For now, I'm assuming, one of them causes the error, which is:

```
[69/248] Linking C shared library zuluCrypt-cli/libzuluCrypt.so.1.2.0
FAILED: [code=1] zuluCrypt-cli/libzuluCrypt.so.1.2.0 
: && /usr/bin/cc -fPIC -DNDEBUG -fstack-clash-protection -D_FORTIFY_SOURCE=2 -mtune=generic -O2    -ffile-prefix-map=/builddir/zulucrypt-7.1.0/build=.  -shared -Wl,-z,relro -Wl,-z,now -Wl,--as-needed -pie -Wl,-z,relro -Wl,-z,now  -Wl,--dependency-file=zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/link.d -Wl,-soname,libzuluCrypt.so.1.2.0 -o zuluCrypt-cli/libzuluCrypt.so.1.2.0 zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/bitlocker.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/mountinfo.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/create_mapper_name.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/is_path_valid.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/version.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/close_volume.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/add_key.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/remove_key.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/empty_slots.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/status.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/create_luks.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/is_luks.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/create_volume.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/close_mapper.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/open_luks.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/open_plain.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/open_volume.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/mount_volume.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/unmount_volume.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/user_home_path.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/create_loop_device.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/blkid_evaluate_tag.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/open_tcrypt.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/volume_type.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/mount_fs_options.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/real_path.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/file_path_security.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/create_tcrypt.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/parse_fstab.c.o zuluCrypt-cli/CMakeFiles/zuluCrypt.dir/lib/resolve_paths.c.o  zuluCrypt-cli/libString.a  zuluCrypt-cli/libStringList.a  zuluCrypt-cli/libProcess.a  external_libraries/tcplay/libtcplay.a  /usr/lib/libcryptsetup.so  /usr/lib/libblkid.so  /usr/lib/libuuid.so  /usr/lib/libdevmapper.so  -lgcrypt  -pthread  -L/usr/lib64  -ldevmapper  -luuid  -lgcrypt -lgpg-error && :
/usr/bin/ld: /usr/lib64/gcc/x86_64-unknown-linux-gnu/14.3.0/../../../../lib64/Scrt1.o: in function `_start':
/builddir/glibc-2.41/csu/../sysdeps/x86_64/start.S:103:(.text+0x17): undefined reference to `main'
collect2: error: ld returned 1 exit status
[70/248] Building C object zuluCrypt-cli/CMakeFiles/zuluCrypt-exe.dir/bin/add_key.c.o
[71/248] Building CXX object external_libraries/lxqt_wallet/frontend/CMakeFiles/lxqt-wallet.dir/lxqt_wallet.cpp.o
[72/248] Building CXX object external_libraries/lxqt_wallet/frontend/CMakeFiles/lxqt-wallet.dir/lxqt_kwallet-dbus.cpp.o
ninja: build stopped: subcommand failed.
```

Too, me it seems weird, because there is no step where `libzuluCrypt.so.1.2.0 ` gets built, hence it's not surprising, it can't be linked either. There is no such file in zuluCrypt-cli (or elsewhere).

So, not sure what to make of it - or how to fix it myself.

--- a/zuluCrypt-cli/CMakeLists.txt
+++ b/zuluCrypt-cli/CMakeLists.txt
@@ -132,11 +132,11 @@ endif()
         TARGET_LINK_LIBRARIES( zuluCrypt     String StringList Process ${cryptsetup_lib} ${blkid} ${uuid_lib} ${devmapper_lib} -lgcrypt -lzuluplay )
  endif()
 
- set_target_properties( zuluCrypt PROPERTIES LINK_FLAGS "-pie -Wl,-z,relro -Wl,-z,now" )
+ set_target_properties( zuluCrypt PROPERTIES LINK_FLAGS "-Wl,-z,relro -Wl,-z,now" )
 
  TARGET_LINK_LIBRARIES( zuluCrypt-exe zuluCrypt zuluCryptPluginManager )
 
- set_target_properties( zuluCrypt-exe PROPERTIES LINK_FLAGS "-pie -Wl,-z,relro -Wl,-z,now" )
+ set_target_properties( zuluCrypt-exe PROPERTIES LINK_FLAGS "-Wl,-z,relro -Wl,-z,now" )
 
  #TARGET_LINK_LIBRARIES( zuluCrypt-static String StringList Process )
  #TARGET_LINK_LIBRARIES( zuluCrypt-exe-static zuluCrypt-static zuluCryptPluginManager-static )

